apps/api/src/vault/dto/create-vault.dto.ts

```diff
@@ -0,0 +1,75 @@
+import { ApiProperty } from '@nestjs/swagger'
+import {
+  Currency,
+  Donation,
+  Expense,
+  Prisma,
+  RecurringDonation,
+  Transfer,
+  Withdrawal,
+} from '@prisma/client'
+import { Expose } from 'class-transformer'
+import { IsOptional, IsString, IsUUID } from 'class-validator'
+
+@Expose()
+export class CreateVaultDto {
+  @Expose()
+  @ApiProperty({ enum: Currency })
+  currency: Currency
+
+  @Expose()
+  @IsString()
+  @ApiProperty()
+  name: string
+
+  @Expose()
+  @ApiProperty()
+  @IsString()
+  @IsUUID()
+  campaignId: string
+
+  @Expose()
+  @ApiProperty()
+  @IsOptional()
+  expenses?: Expense[]
+
+  @Expose()
+  @ApiProperty()
+  @IsOptional()
+  sourceTransfers?: Transfer[]
+
+  @Expose()
+  @ApiProperty()
+  @IsOptional()
+  targetTransfers?: Transfer[]
+
+  @Expose()
+  @ApiProperty()
+  @IsOptional()
+  donations?: Donation[]
+
+  @Expose()
+  @ApiProperty()
+  @IsOptional()
+  withdraws?: Withdrawal[]
+
+  @Expose()
+  @ApiProperty()
+  @IsOptional()
+  recurringDonations?: RecurringDonation[]
```

Comment:
When the vault is created these will always be empty, so please remove them from the create-vault.dto. They will happen through the operations of Donation and making Expenses/Withdrawal later.